nomad/vault.go

```diff
@@ -429,7 +429,7 @@ OUTER:
 				retryTimer.Reset(v.config.ConnectionRetryIntv)
 				v.l.Lock()
 				v.connEstablished = true
-				v.connEstablishedErr = fmt.Errorf("Nomad Server failed to establish connections to Vault: %v", err)
+				v.connEstablishedErr = structs.NewServerSideError(fmt.Errorf("Nomad Server failed to establish connections to Vault: %v", err))
```

Comment:
It'd be really nice if Vault didn't reformat API errors into one giant string in the `api` package :(

https://github.com/hashicorp/nomad/blob/89845b3b516b870b7953380789d58f98da314726/vendor/github.com/hashicorp/vault/api/response.go#L27-L67

If that just returned `fmt.Errorf(bodyBuf.String())` our error messages would be much nicer! No embedded newlines and useless HTTP metadata (url+status).

Ideally Vault would return a custom error type that allowed us to pluck out the bits of metadata we cared about.

No need to block this PR on that, but do you think we should open an issue with them?